pytorch_lightning/core/lightning.py

```diff
@@ -62,17 +63,22 @@ def __init__(self, *args, **kwargs):
         #: True if using ddp2
         self.use_ddp2 = False
 
+        # True if on tpu
+        self.use_tpu = False
+
         #: True if using amp
         self.use_amp = False
 
-        self.hparams = None
-
         #: Current dtype
         self._dtype = torch.float
 
         #: device reference
         self._device = torch.device('cpu')
 
+        self.module_arguments = []
```

Comment:
@Borda ?

Reply:
no, it is list of names... in past assigning pointer to `hparams` dict was safe that it always has the players values, but with mapping it to `self`and any later edit would break consistency so in `self.module_arguments` you would have the init values abut not the actual in time of saving...
now when you ask for `self.hparams` you have list of names and all values are collected JIT